crates/experimentation_client/src/lib.rs

```diff
@@ -265,37 +261,39 @@ async fn get_experiments(
         sort_on: None,
         sort_by: None,
     };
+    let pagination_params = PaginationParams::all_entries();
+    let endpoint = format!("{hostname}/experiments?{pagination_params}&{list_filters}");
+    let experiment_response = http_client
+        .get(endpoint)
+        .header("x-tenant", tenant.to_string())
+        .header("If-Modified-Since", start_date.to_rfc2822())
+        .send()
+        .await
+        .map_err_to_string()?;
 
-    loop {
-        pagination_params = PaginationParams {
-            page: Some(page),
-            ..pagination_params
-        };
-        let endpoint =
-            format!("{hostname}/experiments?{pagination_params}&{list_filters}");
-        let list_experiments_response = http_client
-            .get(endpoint)
-            .header("x-tenant", tenant.to_string())
-            .send()
-            .await
-            .map_err_to_string()?
-            .json::<PaginatedResponse<ExperimentResponse>>()
-            .await
-            .map_err_to_string()?;
-
-        let experiments = list_experiments_response.data;
-
-        for experiment in experiments.into_iter() {
-            curr_exp_store.insert(experiment.id.to_string(), experiment);
-        }
-        if page < list_experiments_response.total_pages {
-            page += 1;
-        } else {
-            break;
+    match experiment_response.status() {
+        StatusCode::NOT_MODIFIED => {
+            return Err(format!(
+                "{} EXP: skipping update, remote not modified",
+                tenant
+            ));
         }
-    }
+        StatusCode::OK => log::info!(
+            "{}",
+            format!("{} EXP: new config received, updating", tenant)
+        ),
```

Comment:
Do this

```suggestion
        StatusCode::OK => log::trace!("{} EXP: new config received, updating", tenant),
```

Nitpick
Also the log line would be more useful if it contained the number of changes or something

Reply:
Let it be log::info , any reason for trace ?